install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -217,6 +217,10 @@ data:
   # Only effective when monitor aggregation is set to "medium" or higher.
   monitor-aggregation-flags: {{ .Values.bpf.monitorFlags }}
 
+{{- if .Values.bpf.compileDebug }}
+  # Enable debugging of the BPF compilation process.
+  bpf-compile-debug: "true"
```

Comment:
I'm going to impersonate @aanm : I know this is not related to your PR, but can't we move this flag to `--debug-verbose` which would reduce cilium-agent flags count by one?

Reply:
I would keep these options separated (at least for now) as:
* my understanding is that they provide debug visibility into different areas of the datapath logic
* `--bpf-compile-debug` may not work on a setup that does not use the Cilium docker images (because of potentially unsupported versions of iproute2/llvm)

so I can see how a user may want to enable `--debug-verbose=datapath` without having the debug symbols in their BPF program

Reply:
I'm going to impersonate myself:
We could have `--debug-verbose=datapath` and `--debug-verbose=datapath-compile` for without and with debug symbols in their BPF program.

Reply:
Right, sorry, I assumed you wanted to move this into `--debug-verbose=datapath`

Reply:
Is there any reason we shouldn't enable this by default? I doubt anyone is running Cilium with their own version of iproute2 or LLVM (the first because our fork is needed, the second because it would generate too many verifier issues).

Reply:
> I doubt anyone is running Cilium with their own version of iproute2 or LLVM (the first because our fork is needed, the second because it would generate too many verifier issues).

Even if that's the case for the end users (I wasn't aware of that but it's good to know) that may not be true in the context of development/testing. Are we fine also in that context to have a single option which controls both debug features of the datapath as well as generation of debug information? (Genuine question as I'm not opposed to that)

Reply:
Just to clarify: I was not suggesting a single option but either 1) a lack of option and `-g` always enabled or 2) an option to opt-out from `-g`.

Reply:
OK now I understand :+1:
I would avoid option 1) for the the concerns I expressed in the previous comment.
For option 2) I think that could work but it's probably not going to fit into `--debug-verbose` (so for this we would need to ~add another option~ replace `--bpf-compile-debug` with something else).

@aanm @brb I'd like to hear your thoughts as well